.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,92 @@
+name: Linux
+
+on: [push, pull_request]
+
+jobs:
+  TestLinux:
+    name: Python ${{ matrix.python }} (GEOS ${{ matrix.geos }}, numpy ${{ matrix.numpy || 'latest' }})
+    # runs-on: ubuntu-latest
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # 2015
+          - python: 3.5
+            geos: 3.5.2
+            numpy: 1.10.4
+          # 2017
+          - python: 3.6
+            geos: 3.6.4
+            numpy: 1.13.3
+          # 2018
+          - python: 3.7
+            geos: 3.7.3
+            numpy: 1.15.4
+          # 2019
+          - python: 3.8
+            geos: 3.8.1
+            numpy: 1.17.5
+          # 2020
+          - python: 3.9
+            geos: 3.8.1
+            numpy: 1.19.4
+          # dev
+          - python: 3.9
+            geos: master
+
+    env:
+      GEOS_VERSION: ${{ matrix.geos }}
+      GEOS_INSTALL: ${{ github.workspace }}/geosinstall/geos-${{ matrix.geos }}
```

Comment:
I wasn't able to figure out how to use $HOME here for `GEOS_INSTALL`. Now in practice, I suppose it doesn't matter that it is put into the workspace dir (that is where your repo is cloned). 
(only needed to ensure that the cache path is also using this directory)